### DISCUSSION TITLE
内网nas搭建一个xmake repo，是需要怎么样的一个完整过程？

### DISCUSSION THREAD

**Discussion by ThzShun** [xmake-io/xmake-repo#2390] [General]
大家好，长期开发以来，很需要一个c++的包管理器。
诸多考量和尝试，觉得xmake非常有前景，而且极具人性化。

由于刚学xmake没几天，但是又需要快速搭建起一个能够分发二进制包的包管理器。

公司内网仅有一个nas设备，然后已经在上面装了gitea服务，具备一般的git仓库功能。
想问在这个基础上，如何搭建xmake-repo和xmake-artifacts？以达到同事间使用包，能够add_packages就完事（并且需要看不到对应的包源码）。

目前有几个问题。
1. gitea创建了一个独立账户和仓库用于xmake-repo，如 https://ip/xmake-org/xmake-repo.git。
那么如何去维护这个repo.git？
```
包的维护人员，通过`xmake package -f remote`，生成packages，修改相关urls、versions信息，
然后拉取xmake-repo.git, 复制上述生成的packages，在推送吗？
```

2. 二进制artifacts又该怎么维护？
构建环境众多、编译参数又多样，两者复合起来，能产生众多二进制包，如何去维护呢？
还有上述1步骤中，涉及到的urls和sha256 等等，又该如何去一致的维护？

**Comment by waruqi** [comment]
参考 https://github.com/xmake-io/xmake-repo 和 文档。。https://xmake.io/#/zh-cn/package/remote_package?id=%e4%bd%bf%e7%94%a8%e8%87%aa%e5%bb%ba%e7%a7%81%e6%9c%89%e5%8c%85%e4%bb%93%e5%ba%93

二进制 artifacts 目前仅支持 windows/msvc 参考 https://github.com/xmake-mirror/build-artifacts

linux下 artifacts ，也可以参考这个自己搞，有人跑通过，但目前官方暂时没提供支持

**Reply by ThzShun** [reply]
现在能够找到包了，但是install失败了。我看了圈文档和xmake-repo的一些实例包，都没发现`解压`这个动作，也就是download回来的是一个压缩包（7z或zip），比如`libusb`，on_install这个lua回调函数是不是会前置执行解压操作？

**Reply by waruqi** [reply]
会自动解压

**Comment by ThzShun** [comment]
已成功搭建，由于gitea的action还不完善，许多操作还是手动完成。
